.gitlab/ci/on-push.yml

```diff
@@ -7,6 +7,25 @@
   - echo "======= Download configuration ========"
   - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
 
+trigger-private-build:
+  needs: []
+  stage: unittests-and-validations
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push)$/ && $SKIP_PRIVATE_BUILD != "true"'
+    - if: '$NIGHTLY && $SKIP_PRIVATE_BUILD != "true"'
+    - if: '$INSTANCE_TESTS && $SKIP_PRIVATE_BUILD != "true"'
+    - if: '$DEMISTO_SDK_NIGHTLY && $SKIP_PRIVATE_BUILD != "true"'
+  script:
+    - |
+      if [ $CI_COMMIT_BRANCH =~ pull/[0-9]+ ]; then
```

Comment:
Seems like you could just add this condition to the `rules` section as the first rule, e.g.
```
    - if: '$CI_COMMIT_BRANCH =~ pull/[0-9]+'
      when: never